### ISSUE TITLE
Timer: Add foreign language triggers

### ISSUE THREAD

**Issue by bsstoner** [duckduckgo/zeroclickinfo-spice#3119]
User feedback:

**temporizador**
[https://duckduckgo.com/?q=temporizador&kp=1](https://duckduckgo.com/?q=temporizador&kp=1)
_"Please add Timer IA: https://duckduckgo.com/?q=countdown"_

------
IA Page:  http://duck.co/ia/view/timer 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mattr555

**Comment by mattr555** [comment]
We don't usually do this for other languages or other IAs, should this be closed in favor of a broader i18n plan? @moollaza

**Comment by moollaza** [comment]
@bsstoner I think this needs much more consideration. If we're going to add foreign triggers for 1, why wouldn't we do it for all? That will create an inconsistent experience for non-english speaker users IMO.

Also, if it triggers in the user's language, I think the expectation would be for the answer to be in that language as well? If a non english user enters a non-english query that shows non-english organics, but an english timer, it would look odd.

Also, much of the code around IAs is written with regex and logic to parse english words/phrases/characters. How are we going to deal with 1 or more other languages for those IAs?

**Comment by bsstoner** [comment]
@moollaza I don't think it has to be all or nothing thing. We have to start somewhere. If we wait for a perfect solution that covers every IA or every language I don't think it will ever happen.

In the short-term, I think it can be a relatively easy problem if you limit the scope to the handful of IA's that are preventing people in other countries from adopting DDG (Weather, Time, Currency, Conversions, Calculator, Timer) and the few languages that the majority of users use (DE, FR and ES). We've already started to i18n weather and it's allowed us to address a lot of complaints people had with using DDG internationally.

Why can't we expand the regex here to trigger in German, French and Spanish and add the handful of text tokens needed to translate the UI?

**Comment by moollaza** [comment]
> Why can't we expand the regex here to trigger in German, French and Spanish and add the handful of text tokens needed to translate the UI?

@bsstoner if we have parameters for what we want (i.e. specific IAs and languages) then I think this is much more acceptable, because it's contained.

However, I think we should plan how we want to handle/organize the non-english code though. E.g. defining French, German, Spanish regex in separate modules. Hopefully an approach like that could keep the Perl modules uncluttered.

**Comment by jbarrett** [comment]
I don't think we should wait for some notional i18n effort in IA infrastructure if we can get quick wins now with simple inline additions, especially if, as @bsstoner suggests, we limit this to a strict highly-popular subset.

Eventually, when things are done The Right Way™, this set would be easy to refactor, right?

**Comment by MichaelRupertDev** [comment]
I'd be happy to work on this, just want to make sure the trigger words are accurate, could we get a list of international trigger words?

**Comment by ofekdeitch** [comment]
I would like to help with **Hebrew** support :grin:

**Comment by gvklive** [comment]
I would be happy to work on this. It's my first time though... So, i would need someone to guide me a little towards the path.

**Comment by caco13** [comment]
I can help with brazilian portuguese translations.

**Comment by husseinraoouf** [comment]
I could help with Arabic translations

**Comment by gvklive** [comment]
If no one else is working, I think I'll start working on this. Just give me a go. & I'll be in touch for any help. sounds good?

**Comment by GiovanniSM20** [comment]
me too @caco13

**Comment by c-Rolland** [comment]
I would like to help with Hungarian support

**Comment by ValeriaVG** [comment]
Greetings!
I started to work on this issue and bumped into issues with non-ASCII character triggers.

There are no problems on adding **temporizador** as in the description of the issue, but **chrônometro** 
gives a ```[NOTICE] Detected potential UPPERCASE triggers ```. Same for **таймер**.

![temporizador](https://user-images.githubusercontent.com/5821189/28646152-f9a9580e-7268-11e7-8e67-4fc7c96b79ed.png)

So, I believe non-ASCII characters recognition should be the first step towards implementing multilingual support for IA.

The next problem is that in some languages ( Russian for example),  the noun that follows the number changes based on the number: 

- 1 секунда (1 second)
- 2 секунды (2 seconds)
- 5 секунд  (5 seconds)

And the last thing to do is to pass the language to the template, so it can be translated to the proper language.

I'd gladly help will all of it and translations to Russian, Portuguese, French and Spanish, but I'm afraid I'm completely unfamiliar to perl. Can learn though, but I'd prefer Front-End.

P.S.I think it would be better to have a shared component to take care of translations, numbers and languages so it would be easier (and way DRYer :smile:) to translate other IA.

**Comment by PIROMANN** [comment]
I can help with translate to Russian/Ukrainian, I am familiar with those langunages. If it is needed - just let know
